### PR TITLE
docs: websocket documentation

### PR DESCRIPTION
**Pull request by ruancomelli**
# Issue Number: #480 

# Body

## Summary by CodeRabbit

- **New Features**
  - Added documentation on using Smart NgRX with WebSockets, including handling updates, creations, and deletions of records.

- **Enhancements**
  - Updated the category assignment for the demo walkthrough documentation.
  - Renamed the constant `CreatePage` to `ExistingSelectorsPage` in the existing selectors documentation.

- **Chores**
  - Upgraded package manager version from `pnpm@8.6.4` to `pnpm@9.3.0`.

### CODE REVIEW

**Comment by ruancomelli** [comment]
This is a benchmark review for experiment `mermaid_diagrams`.
Run ID: `mermaid_diagrams/benchmark_2024-09-20T10-34-10_v1-22-0-133-gbcf338da5-dirty`.

This pull request was cloned from `https://github.com/DaveMBush/SmartNgRX/pull/481`. _(Note: the URL is not a link to avoid triggering a notification on the original pull request.)_
<details>
<summary>Experiment configuration</summary>

```yaml
review_config:
  # User configuration for the review
  # - benchmark - use the user config from the benchmark reviews
  # - <value> - use the value directly
  user_review_config:
    enable_ai_review: true
    enable_rule_comments: false

    enable_complexity_comments: false
    enable_security_comments: false
    enable_tests_comments: false
    enable_comment_suggestions: false
    enable_functionality_review: false

    enable_pull_request_summary: false
    enable_review_guide: true

    enable_approvals: false

  ai_review_config:
    # The model responses to use for the experiment
    # - benchmark - use the model responses from the benchmark reviews
    # - llm - call the language model to generate responses
    model_responses:
      comments_model: benchmark
      comment_area_model: benchmark
      comment_validation_model: benchmark
      comment_suggestion_model: benchmark
      complexity_model: benchmark
      functionality_model: benchmark
      security_model: benchmark
      tests_model: benchmark
      pull_request_summary_model: benchmark
      review_guide_model: llm
      overall_comments_model: benchmark

# The pull request dataset to run the experiment on
pull_request_dataset:
- https://github.com/sourcery-ai/core/pull/4607
- https://github.com/sourcery-ai/core/pull/4631
- https://github.com/sourcery-ai/core/pull/4647
    # CodeRabbit examples:
- https://github.com/2lambda123/-Orange-OpenSource-oorobot/pull/15
- https://github.com/2lambda123/galaxyproject-galaxy/pull/12
- https://github.com/a0v0/avtoolz/pull/79
- https://github.com/adityask98/Hotaru/pull/10
- https://github.com/agdas/vscode/pull/2
- https://github.com/agluszak/hirschgarten/pull/2
- https://github.com/alexsnow348/insightface/pull/46
- https://github.com/alikuxac/utilities/pull/10
- https://github.com/AlphaDev87/timba-api/pull/49
- https://github.com/AngeloTadeucci/Maple2/pull/239
- https://github.com/AngeloTadeucci/Maple2.File/pull/36
- https://github.com/AngeloTadeucci/Maple2/pull/233
  # Examples where CodeRabbit does not generate diagrams
- https://github.com/baptisteArno/typebot.io/pull/1778
- https://github.com/btipling/foundations/pull/33
- https://github.com/btipling/foundations/pull/31
- https://github.com/chintu-777/jaeger/pull/1
- https://github.com/coji/remix-docs-ja/pull/55
- https://github.com/DaveMBush/SmartNgRX/pull/622
- https://github.com/DaveMBush/SmartNgRX/pull/481
- https://github.com/dkittle/party-connections/pull/6
- https://github.com/Drajad-Kusuma-Adi/onstudy-backend/pull/6
- https://github.com/imaami/libcanth/pull/2

# Questions to ask to label the review comments
review_comment_labels: []
# - label: correct
#   question: Is this comment correct?

# Benchmark reviews generated by running
#   python -m scripts.experiment benchmark <experiment_name>
benchmark_reviews:
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4607
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/338
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4631
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/339
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4647
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/340
- dataset_pull_request: https://github.com/2lambda123/-Orange-OpenSource-oorobot/pull/15
  review_pull_request: https://github.com/sourcery-ai-experiments/-Orange-OpenSource-oorobot/pull/2
- dataset_pull_request: https://github.com/2lambda123/galaxyproject-galaxy/pull/12
  review_pull_request: https://github.com/sourcery-ai-experiments/galaxyproject-galaxy/pull/1
- dataset_pull_request: https://github.com/adityask98/Hotaru/pull/10
  review_pull_request: https://github.com/sourcery-ai-experiments/Hotaru/pull/2
- dataset_pull_request: https://github.com/agdas/vscode/pull/2
  review_pull_request: https://github.com/sourcery-ai-experiments/vscode/pull/3
- dataset_pull_request: https://github.com/agluszak/hirschgarten/pull/2
  review_pull_request: https://github.com/sourcery-ai-experiments/hirschgarten/pull/2
- dataset_pull_request: https://github.com/alikuxac/utilities/pull/10
  review_pull_request: https://github.com/sourcery-ai-experiments/utilities/pull/2
- dataset_pull_request: https://github.com/AlphaDev87/timba-api/pull/49
  review_pull_request: https://github.com/sourcery-ai-experiments/timba-api/pull/2
- dataset_pull_request: https://github.com/AngeloTadeucci/Maple2/pull/239
  review_pull_request: https://github.com/sourcery-ai-experiments/Maple2/pull/3
- dataset_pull_request: https://github.com/AngeloTadeucci/Maple2.File/pull/36
  review_pull_request: https://github.com/sourcery-ai-experiments/Maple2.File/pull/2
- dataset_pull_request: https://github.com/AngeloTadeucci/Maple2/pull/233
  review_pull_request: https://github.com/sourcery-ai-experiments/Maple2/pull/4

```

</details>

**Comment by SourceryAI** [comment]
## Reviewer's Guide by Sourcery

This pull request adds documentation for using Smart NgRX with WebSockets, updates existing documentation, and makes some minor changes to improve code organization and naming. The main focus is on explaining how to handle WebSocket interactions for updates, creations, and deletions of records in Smart NgRX.

### File-Level Changes

| Change | Details | Files |
| ------ | ------- | ----- |
| Added new documentation for using Smart NgRX with WebSockets | <ul><li>Created a new markdown file explaining WebSocket integration</li><li>Added examples for handling updates, creates, and deletes</li><li>Explained how to translate server-side table names to client-side representations</li><li>Created a new NgDocPage for the WebSockets documentation</li></ul> | `apps/documentation/src/app/using-smart-ng-rx/web-sockets/index.md`<br/>`apps/documentation/src/app/using-smart-ng-rx/web-sockets/ng-doc.page.ts` |
| Updated existing documentation structure and naming | <ul><li>Changed category assignment for demo walkthrough documentation</li><li>Renamed constant from CreatePage to ExistingSelectorsPage</li><li>Created a new category file for Demo Walkthrough</li><li>Deleted the cookbook category file</li></ul> | `apps/documentation/src/app/demo-walkthrough/tbd/ng-doc.page.ts`<br/>`apps/documentation/src/app/using-smart-ng-rx/using-with-existing-selectors/ng-doc.page.ts`<br/>`apps/documentation/src/app/demo-walkthrough/ng-doc.category.ts`<br/>`apps/documentation/src/app/cookbook/ng-doc.category.ts` |

### Sequence Diagram

```mermaid
sequenceDiagram
    participant Client
    participant Server
    participant SmartNgRX
    Server->>Client: WebSocket message (action, table, ids)
    Client->>SmartNgRX: handleSocketNotification(table, action, ids)
    SmartNgRX->>SmartNgRX: Process notification
    SmartNgRX->>Client: Update state
```

---

<details>
<summary>Tips</summary>

- Trigger a new Sourcery review by commenting `@sourcery-ai review` on the pull request.
- Continue your discussion with Sourcery by replying directly to review comments.
- You can change your review settings at any time by accessing your [dashboard](https://app.sourcery.ai):
    - Enable or disable the Sourcery-generated pull request summary or reviewer's guide;
    - Change the review language;
- You can always [contact us](mailto:support@sourcery.ai) if you have any questions or feedback.

</details>

File: apps/documentation/src/app/using-smart-ng-rx/web-sockets/index.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# WebSockets
+
+To use SmartNgRX with WebSockets, you'll need to establish your own WebSocket connection. The only information the that you'll need to pass up to the client are which type of action needs to be performed, the table name that is impacted, and the ids of the records that are impacted.
```
Comment (SourceryAI) [inline apps/documentation/src/app/using-smart-ng-rx/web-sockets/index.md:3]: **suggestion (documentation):** Remove extra 'the' in the sentence

The sentence should read: 'The only information that you'll need to pass up...'

File: apps/documentation/src/app/using-smart-ng-rx/web-sockets/index.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+## Updates
+
+So, as an example, if you update the content of a row in the `users` table, you'll need to send a message to the client that might like this:
```
Comment (SourceryAI) [inline apps/documentation/src/app/using-smart-ng-rx/web-sockets/index.md:7]: **suggestion (documentation):** Fix typo in 'might like this'

Change to 'might look like this'

File: apps/documentation/src/app/using-smart-ng-rx/web-sockets/index.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+```json
+{
+  "action": "update",
+  "table": "users",
+  "ids": ["1"]
+}
```
Comment (SourceryAI) [inline apps/documentation/src/app/using-smart-ng-rx/web-sockets/index.md:14]: **suggestion (documentation):** Add comments to explain the switch statement cases

Consider adding brief comments for each case to explain what each transformation does and why it's necessary.

**Review by SourceryAI — COMMENTED** [review]
Hey @ruancomelli - I've reviewed your changes - here's some feedback:

**Overall Comments**:
- Could you provide more context on the removal of the cookbook category? Was this intentional, or is it being replaced by the demo walkthrough?

<details>
<summary>Here's what I looked at during the review</summary>

- 🟢 **General issues**: all looks good
- 🟡 **Documentation**: 3 issues found
</details>

***

[LangSmith trace](https://smith.langchain.com/o/7d8518ad-6dd3-5c7b-96d2-757769296335/projects/p/88560225-5072-48bc-967c-dc54da53f265/r/bfa04b61-c462-4b80-9bc6-8fa0b22edbe6?poll=true)

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment to tell me if it was helpful.
</sub>

**Pull request closed without merging**